src/components/ContactDetails/ContactDetailsAvatar.vue

```diff
@@ -118,6 +122,7 @@ export default {
 			root: generateRemoteUrl(`dav/files/${getCurrentUser().uid}`),
 			width: 0,
 			height: 0,
+			facebookid: 0,
```

Comment:
If we want to support more types in the future, I would suggest we don't go for specified variables.
I would keep a list of available social avatar provider in the php sync service and expose this list to the vue app with an initial state.

Then we can directly have a computed here that filter all existing social ids here based on this list, that would be cleaner and much more easier to manage.
That way we just have to add a new php sync method for twitter (for example) and the ui will automatically do the job, seems good to you? :smiley:

Reply:
Would that work with social networks like mastodon?

Reply:
I don't know Mastodon, but if you can generate an URL to public available data, it should be possible. So gravatar would be an easy task, for example.

Reply:
libravatar also.
The problem with mastodon is that each user could have another instance with a different url.

Reply:
That shouldn't be a problem if you can find this information in the contact's data.

Reply:
> I found something really cool:
> https://avatars.io/ !
> Couldn't find much information about that, but look:
> https://avatars.io/facebook/4
> https://avatars.io/facebook/zuck
> https://avatars.io/twitter/Nextclouders

It seems avatars.io has been out there for a bit and is quite established (in smaller projects only). On the other hand the domain expires in a month.

> I don't know who own them, it can lead to undesired behaviour :)
> If they decide to only return awful images to annoy everyone, this will get back to us disappointed

So I'll remove it then. Too bad!

@SuperSandro2000 : is it possible to access Mastodon profile pictures from the public?

Reply:
Let's start with facebook and tumblr for now. But feel free to post new (public) urls and I will be happy to add them.

Reply:
@SuperSandro2000 I have added Mastodon support with my latest commit (0527ede) - do you want to test it? profiles can be entered as `@nextcloud@mastodon.xyz`, `https://mastodon.xyz/@nextcloud` or `https://mastodon.xyz/users/nextcloud`.